### PR TITLE
Start using readthedocs

### PR DESCRIPTION
**Pull request by MartinThoma**
See #27 and #199.

The Sphinx documentation is only a draft. If you like it, I can adjust it (e.g. add `converters`, `cursors`, `cursors`, `util`) and probably move the examples from the README to the Sphinx docs.

### CODE REVIEW

**Comment by methane** [comment]
@MartinThoma FWIW, do you know `--force-with-rease` option of `git pull` command?
No reason to create new pull request after rebase.

File: docs/requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+pymysql
```
Comment (methane) [inline docs/requirements.txt]: Why do you install pymysql from PyPI instead of using current repository?

**Comment by MartinThoma** [comment]
@methane No, I didn't know that. Thank you.

**Comment by methane** [comment]
See http://docs.readthedocs.io/en/latest/faq.html#can-i-document-a-python-package-that-is-not-at-the-root-of-my-repository

**Comment by methane** [comment]
@MartinThoma mention me when review ready.

**Comment by methane** [comment]
@MartinThoma Did you check admin page of readthedocs?
"Install your project inside a virtualenv using setup.py install" may help you.

File: docs/requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+git+git://github.com/PyMySQL/PyMySQL.git
```
Comment (methane) [inline docs/requirements.txt]: `..` is enough, maybe.

Comment (methane) [inline docs/requirements.txt]: Or use "Install your project inside a virtualenv using setup.py install".

**Comment by MartinThoma** [comment]
@methane The documentation draft is ready so that one can get a feeling for how it might look like when everything would be done this way. If you are interested in getting this documentation, then (and only then) I will do so for the other modules. Are you interested?

**Comment by methane** [comment]
@MartinThoma Having RTD is nice.  But don't use pymysql.readthedocs.io for preview.
Unless you add me to the project admin of RTD, I can't merge this pull request and I can't use pymysql.readthedocs.io domain.

**Comment by methane** [comment]
BTW, I already have readthedocs for mysqlclient
https://mysqlclient.readthedocs.io/en/latest/#

There are no need of `docs/requirements.txt`.  Please see my comment above.

**Comment by methane** [comment]
Don't strip information from README!!
README should provide quick and **complete** information.
Duplication between README and RTD is not a problem.

**Comment by MartinThoma** [comment]
@methane I've added you as an admin on readthedocs.

> Don't strip information from README!!
README should provide quick and complete information.
Duplication between README and RTD is not a problem.

Are you sure you want it like this? Because then you might have to change always things in two places and it also makes the README longer.

**Comment by methane** [comment]
> I've added you as an admin on readthedocs.

Thanks.

> Are you sure you want it like this? Because then you might have to change always things in two places and it also makes the README longer.

Yes.  Despite having doc in another site, README should provides enough information for people
choosing library they use.

Information for developer can be removed from README.  It includes:

* Install from git checkout
* Running test suite
* Link for MySQL protocol

Other information (Requirements, install from PyPI, Link to DB-API, MySQL, ML and license) should be in README.

**Comment by methane** [comment]
Thanks.  I've merged and organized the docs.

**Comment by MartinThoma** [comment]
Nice :-) I'm happy I could help.

**Pull request merged**